tests/test_admin.py

```diff
@@ -2526,38 +2527,79 @@ class ListActionsTestCase(CMSTestCase):
-        edit_endpoint = reverse("admin:djangocms_versioning_pollcontentversion_edit_redirect", args=(version.pk,),)
-        response = func(menu_content)
+        actual_enabled_control = self.modeladmin._get_edit_link(
+            version.content, request, disabled=False
+        )
+        expected_enabled_state = (
+            '<a class="btn cms-versioning-action-btn js-versioning-action"'
+            ' href="%s" title="Edit">'
+        ) % draft_edit_url
 
-        self.assertIn("cms-versioning-action-btn", response)
-        self.assertIn('title="Edit"', response)
-        self.assertIn(edit_endpoint, response)
+        self.assertIn(expected_enabled_state, actual_enabled_control)
 
-    def test_edit_link_inactive(self):
+    def test_edit_action_link_disabled_state(self):
         """
-        The edit link should not be shown for a user that does not have the edit permission.
+        Un-editable content models render the edit endpoint as inactive, without a URL
         """
-        content_model = factories.BlogContentWithVersionFactory()
-        version = content_model.versions.last()
-        request = self.get_request("/")
-        request.user = self.get_staff_user_with_no_permissions()
+        version = factories.PollVersionFactory(state=constants.DRAFT)
+        request = RequestFactory().get("/admin/polls/pollcontent/")
+        request.user = factories.UserFactory()
+
+        actual_disabled_control = self.modeladmin._get_edit_link(
+            version.content, request, disabled=True
+        )
+        expected_disabled_control = (
+            '<a class="btn cms-versioning-action-btn inactive" title="Edit">'
+        )
 
-        func = self.modeladmin._list_actions(request)
-        edit_endpoint = reverse("admin:djangocms_versioning_blogcontentversion_edit_redirect", args=(version.pk,),)
-        response = func(version.content)
+        self.assertIn(expected_disabled_control, actual_disabled_control)
+
+    def test_versions_list_action_active(self):
+        """
+        Version list admin action is rendered when enabled, with appropriate URL
+        """
+        draft_version = factories.PollVersionFactory(state=constants.DRAFT)
+        versionable = _cms_extension().versionables_by_content[draft_version.content.__class__]
+        proxy = versionable.version_model_proxy
+        manage_versions_url = admin_reverse(
+            "{app}_{model}_changelist".format(
+                app=proxy._meta.app_label,
+                model=proxy._meta.model_name
+            )
+        )
+        request = RequestFactory().get("/admin/polls/pollcontent/")
+        request.user = factories.UserFactory()
+
+        actual_enabled_manage_versions_control = self.modeladmin._get_manage_versions_link(
+            draft_version.content, request,
+        )
+        expected_enabled_manage_versions_control = \
+            '<a\n    title="Manage versions"\n    class="btn\n\n    cms-versioning-action-btn\n'
```

Comment:
Thos formatting shows us that the template is not using spaceless correctly. We shouldn't have new lines output as this is what the source of the page will look like to (really poorly constructed). 

spaceless and endspaceless template tags will resolve this issue.